src/Components/Misc/FileUpload.vue

```diff
@@ -54,145 +54,124 @@
+                for (j = 0, len = ref.length; j < len; j++) {
+                    thumbnailElement = ref[j];
+                    thumbnailElement.alt = file.name;
+                    thumbnailElement.style.backgroundImage = 'url("' + dataUrl + '")';
                 }
+                return setTimeout((() => file.previewElement.classList.add("dz-image-preview")), 1);
             }
-        },
-
-        methods: {
-
-            /**
-             * Cancel the upload.
-             */
-            cancel()
-            {
-                this.$emit('cancel')
-            },
-
-            /**
-             * Generate the thumbnail.
-             *
-             * @param file
-             * @param dataUrl
-             * @returns {number}
-             */
-            generateThumbnail(file, dataUrl)
-            {
-                let j, len, ref, thumbnailElement;
-
-                if (file.previewElement) {
-                    file.previewElement.classList.remove("dz-file-preview");
-                    ref = file.previewElement.querySelectorAll("[data-dz-thumbnail-bg]");
-                    for (j = 0, len = ref.length; j < len; j++) {
-                        thumbnailElement = ref[j];
-                        thumbnailElement.alt = file.name;
-                        thumbnailElement.style.backgroundImage = 'url("' + dataUrl + '")';
-                    }
-                    return setTimeout(((function(_this) {
-                        return function() {
-                            return file.previewElement.classList.add("dz-image-preview");
-                        };
-                    })(this)), 1);
-                }
-            },
-
-            /**
-             * Handle the error event.
-             */
-            handleError(file, message, xhr)
-            {
-                if (! xhr) {
-                    return;
-                }
+        }
 
-                let response = JSON.parse(xhr.response);
-
-                this.errorMessage = response.errors[0].title;
-            },
-
-            /**
-             * Handle the success event.
-             */
-            handleSuccess(file, response)
-            {
-                this.$store.commit('files/ADD', response);
-                this.$emit('uploaded', response);
-            },
-
-            /**
-             * Upload the files.
-             */
-            upload()
-            {
-                this.$refs.dropzone.processQueue();
+        /**
+         * Handle the error event.
+         */
+        handleError(file: any, message: string, xhr: any)
+        {
+            if (! xhr) {
+                return;
             }
-        },
+
+            let response = JSON.parse(xhr.response);
+
+            this.errorMessage = response.errors[0].title;
+        }
+
+        /**
+         * Handle the success event.
+         */
+        handleSuccess(file: any, response: any)
+        {
+            this.$store.commit('files/ADD', response);
```

Comment:
I'm not sure about this particular line. Do we need to import Files module from our application here?

Reply:
I mean add something like `import { File } from '../../../..App/State/Models';` and then do `File.insert(response);`